### PR TITLE
ci: migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by stevenjoezhang**
Switch to GitHub actions. Automated testing is more efficient than the manual review we currently use.

See also https://github.com/hexojs/site/issues/837 https://github.com/hexojs/site/pull/1458

### CODE REVIEW

File: scripts/validate.js [hunk 1]
```diff
@@ -6,51 +6,71 @@ const { join } = require('path');
...
-    name = String(theme.name).toLocaleLowerCase();
-    if (themeNames.indexOf(name) !== -1) {
-      isValidationPassed = false;
-      break;
+    const name = theme.name.toLocaleLowerCase();
+    if (themes.has(name)) {
+      duplicate.add(name);
     } else {
-      themeNames.push(name);
+      themes.add(name);
```
Comment (SukkaW) [inline scripts/validate.js:30]: Set will be deduped automatically, no need to wrap it in `else`. But it is fine.

File: scripts/validate.js [hunk 2]
```diff
@@ -6,51 +6,71 @@ const { join } = require('path');
 
 const { listDir } = require('hexo-fs');
 const { log } = hexo;
-const sizeOf = require('image-size');
+const sharp = require('sharp');
```
Comment (SukkaW) [inline scripts/validate.js:9]: Sharp requires node-gyp to build. Better enable node_modules cache in the CI

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: scripts/validate.js (hunk 2 above)
Comment (stevenjoezhang) [reply scripts/validate.js:9]: Caching `node_modules` may [cause the test to fail](https://github.com/hexojs/hexo/pull/4630#issuecomment-782555535) and [is not recommended](https://github.com/hexojs/hexo/pull/4630#issuecomment-782555535)

`sharp` is also used in `scripts/responsive.js`, I believe this modification will not worsen the running time of the installation script

**Comment by stevenjoezhang** [comment]
TODO: `hexo.locals.get('data').themes` is undefined when running `hexo clean`

**Review by SukkaW — APPROVED** [review]

**Pull request merged**